### DISCUSSION TITLE
Device constantly disconnects/reconnects (when on wireless charger?)

### DISCUSSION THREAD

**Discussion by abbv2x** [mKeRix/room-assistant#436] [Q&A]
Hi all, 

I have a Samsung S10 and a rather simple Home Assistant/Room Assistant automation that turns my entrance lights on when I get home and turns the hall light off when I leave. The automation works great other than one hiccup: there are occasions when the hall lights turn off (or on) randomly when I am home (off, roughly 10 seconds, on/vice versa), most notably when my phone is on a wireless charger and of course when I'm trying to sleep!! My guess is that the charger is affecting the BT signal causing Room Assistant to "think" I'm not home. I have played with the RSSI strength but that doesn't seem to solve the issue. I live in a condo, HA is running on a RPi 4 and the wireless charger in the kitchen about 10 feet away from the RPi with almost direct LOS (Line Of Sight). Anyone else run into this issue?

A solution I'm thinking of is using Node Red to run a timer in parallel with the lights turning on (or off) where if the timer is running (say 5 minutes) so that if the state changes within that 5 minutes, the timer is reset. Once I leave the house, the timer will expire and the 'not_home' state will be published to the lights, thus turning them off. Still mucking around with this. I'm not sure how to make the lights turn on right away but be "blocked" if the timer is running.

Thanks for any help here :)

**Comment by mKeRix** [comment]
So, if I understood that correctly you want to achieve:

- state going to not_home triggers a 5 minute grace period where we wait if it switches back to home
- state going to home has two possible paths
  - timer running? -> block the message, stop timer
  - timer not running? -> turn on your entrance lights as normal

I think you can achieve this via a stoptimer and a variable. On not_home states always start the timer and then set a flow variable like presence_grace_period = true. If the timer expires run your going away flow and set the variable to false. For home states you can use a switch node to check the variable state -> if it's true stop the timer and set it to false, if it's false run your coming home flow. At least that's what I'm imagining in my head right now, there may be more elegant ways. :)

There is nothing native in room-assistant to handle this particular use case, but it is something worth thinking about for the future since it is likely a somewhat common scenario.

**Reply by abbv2x** [reply]
Thanks for the reply. Your review of what I'm trying to achieve is correct. I posted in here to see if anyone has ran into this (both having the same constant disconnect/reconnect issues as well as if there were any in-app solutions, no worries if there wasn't) I was able to get this working after a few iterations within Node Red and it turns out it's pretty simple actually. Note, as an electrician I'm good with logic, not with programming haha!  

I'll summarize in simple terms for others looking to replicate this:
For the lights turning on, it's business as usual. 

- Create an EVENTS STATE node that looks for your phone to connect to room assistant
- This is tied to a CALL SERVICE node that turns on your lights
When the phone connects, the lights are on right away. This is particularly good (no delay that is....) if your wifi signal doesn't reach super far from your entrance door.

- In a new sequence, create another EVENTS STATE node that looks for the phone to be away (not_home) but here's where you set the FOR to NUMBER and a value of 5 minutes.
- This is tied to a CALL SERVICE node that turns your lights off.
As the phone reconnects after a disconnect, the second events state is canceled as the state is no longer "not_home".

Your method above will most certainly work but by doing it this way, the events state looks for the not_home state to remain constant for 5 minutes by itself. I found that by using the stoptimer, I was unable to stop the stop timer although after I got this working I did manage to find the "payload:reset" which stop the timer by youtube hunting. Better late than never I suppose.

Thanks again for the room assistant app and the troubleshooting help! If I may add some points that I was thinking of:
1. Perhaps if it's possible, can you add a way to view the RSSI strength? There are many ways for seeing the wifi strength, but I wasn't able to find any for bluetooth. This would have allowed me to see if it's simply a strength issue (and what value would work to set it at to not drop) or some other issue.
2. As someone who is relatively new to the Home automation world, there are things that I did not know at first and one of those was the fact that your phone has different MAC ID's for wifi and bluetooth. An update to the documentation that denotes BLUETOOTH MAC ID may save some frustration wondering why the phone won't connect (Yes, that's me).
 
Thanks again!
Adam

**Reply by mKeRix** [reply]
You can view all the relevant values from the [room-assistant API](https://www.room-assistant.io/guide/api.html)! I know it is not the easiest way and I want to make that a better experience in the future. It's just a bit further down on the backlog, as there are a few other features that I wanted to get to first.

And thanks for the note about the docs. I've saved a bookmark of this discussion on GitHub, so I'll remember to update this in the future.
